src/ProgressMeter.jl

```diff
@@ -840,6 +840,32 @@ function showprogressdistributed(args...)
     end
 end
 
+function showprogressthreads(args...)
+    progressargs = args[1:end-1]
+    expr = args[end]
+    loop = expr.args[end]
+    r = loop.args[1].args[end]
+    p = gensym()
+    n = gensym()
+
+    ex = quote
+        $n = Int(round(length($(esc(r))) / Threads.nthreads()))
+        global $p
+        $p = ProgressMeter.Progress($n; $(showprogress_process_args(progressargs)...))
+        $(esc(expr))
+        ProgressMeter.finish!($p)
+    end
+
+    update = quote
+        if Threads.threadid() == 1
+            ProgressMeter.next!(ProgressMeter.$p)
```

Comment:
there is an example in the readme with a `Threads.@threads` that updates an all threads, can you try to make it work? The progressbar might be wrong for unevenly distributed tasks

Reply:
Sure thing! I changed the code to update on every thread